examples/examples-cpu/nyc-taxi/xgboost.ipynb

```diff
@@ -86,17 +91,17 @@
    "outputs": [],
    "source": [
     "def prep_df(df: pd.DataFrame) -> pd.DataFrame:\n",
-    "    '''\n",
+    "    \"\"\"\n",
     "    Generate features from a raw taxi dataframe.\n",
-    "    '''\n",
+    "    \"\"\"\n",
     "    df = df[df.fare_amount > 0]  # avoid divide-by-zero\n",
-    "    df['tip_fraction'] = df.tip_amount / df.fare_amount\n",
+    "    df[\"tip_fraction\"] = df.tip_amount / df.fare_amount\n",
     "    \n",
-    "    df['pickup_weekday'] = df.tpep_pickup_datetime.dt.weekday\n",
-    "    df['pickup_weekofyear'] = df.tpep_pickup_datetime.dt.weekofyear\n",
-    "    df['pickup_hour'] = df.tpep_pickup_datetime.dt.hour\n",
-    "    df['pickup_week_hour'] = (df.pickup_weekday * 24) + df.pickup_hour\n",
-    "    df['pickup_minute'] = df.tpep_pickup_datetime.dt.minute\n",
+    "    df[\"pickup_weekday\"] = df.tpep_pickup_datetime.dt.weekday\n",
+    "    df[\"pickup_weekofyear\"] = df.tpep_pickup_datetime.dt.weekofyear\n",
```

Comment:
This might have been the source of some of the warnings: 

```suggestion
    "    df[\"pickup_weekday\"] = df.tpep_pickup_datetime.dt.isotime().day\n",
    "    df[\"pickup_weekofyear\"] = df.tpep_pickup_datetime.dt.isotime().week\n",
```

Reply:
I don't think so. When I tried that, I got this

> AttributeError: 'DatetimeProperties' object has no attribute 'isotime'

Reply:
after trying some stuff though, it does look like you were onto something

> /srv/conda/envs/saturn/lib/python3.7/site-packages/ipykernel_launcher.py:9: FutureWarning: Series.dt.weekofyear and Series.dt.week have been deprecated.  Please use Series.dt.isocalendar().week instead.

Will fix in a followup PR that fixes warnings.

Reply:
That's what I get for trying to remember things :roll_eyes: yes it's `isocalendar().day` and `isocalendar().week`